crates/ruff_linter/src/rules/flake8_pie/rules/unnecessary_spread.rs

```diff
@@ -68,24 +67,27 @@ pub(crate) fn unnecessary_spread(checker: &mut Checker, dict: &ast::ExprDict) {
 fn unnecessary_spread_fix(
     dict: &ast::ExprDict,
     prev_end: TextSize,
-    locator: &Locator,
+    tokens: &Tokens,
 ) -> Option<Fix> {
     // Find the `**` token preceding the spread.
-    let doublestar = SimpleTokenizer::starts_at(prev_end, locator.contents())
-        .find(|tok| matches!(tok.kind(), SimpleTokenKind::DoubleStar))?;
+    let doublestar = tokens
+        .after(prev_end)
+        .iter()
+        .find(|tok| matches!(tok.kind(), TokenKind::DoubleStar))?;
 
     if let Some(last) = dict.iter_values().last() {
         // Ex) `**{a: 1, b: 2}`
         let mut edits = vec![];
         let mut open_parens: u32 = 0;
 
-        for tok in SimpleTokenizer::starts_at(doublestar.end(), locator.contents()).skip_trivia() {
+        for tok in tokens.after(doublestar.end()) {
             match tok.kind() {
-                SimpleTokenKind::LParen => {
+                kind if kind.is_trivia() => {}
+                TokenKind::Lpar => {
```

Comment:
We'll need to be careful here because the `skip_trivia` includes the `Newline` token while the `is_trivia` method on `TokenKind` does not. The latter only includes the `NonLogicalNewline` which is different than `Newline`. It _might_ be that this is ok because the `SimpleTokenizer` emits `Newline` regardless of the context.

Reply:
I think it should be ok to do here because the edits are only generated when inside parentheses which will emit `NonLogicalNewline` instead of `Newline`.

Reply:
Yes